`gulp dist-install` fails because invalid arguments are passed to npm

## Repro Steps

Tested on both MacOS 12.5.1 and Ubuntu 22.04 and the behavior was the same.

```text
~/Downloads node --version
v16.17.0
~/Downloads npm --version
8.19.1
~/Downloads git clone https://github.com/mozilla/pdf.js.git
<output omitted>
~/Downloads cd pdf.js
~/Downloads/pdf.js (master| ✔) npm i --force
<output omitted - it succeeded>
~/Downloads/pdf.js (master|✚1) gulp dist-install
<all steps before the following one succeeded>

### Overwriting all files
[11:28:04] Finished 'createDist' after 2.53 s
[11:28:04] Finished 'dist-pre' after 47 s
[11:28:04] Starting 'createDistInstall'...
Error: command "npm" with parameters "install,build/dist/" exited with code 1
[11:28:05] The following tasks did not complete: dist-install, createDistInstall
[11:28:05] Did you forget to signal async completion?
```

## Related Issues and PRs

- The issue was first reported here: https://github.com/mozilla/pdf.js/issues/15382.
- I submitted a PR to fix this here: https://github.com/mozilla/pdf.js/pull/15428.
- I ran `npm install --force` in the repro steps because of this issue: https://github.com/mozilla/pdf.js/issues/15429

## Full Output
<details>
<summary>Full output:</summary>

```text
~/Downloads node --version
v16.17.0
~/Downloads npm --version
8.19.1
~/Downloads git clone https://github.com/mozilla/pdf.js.git
Cloning into 'pdf.js'...
remote: Enumerating objects: 84821, done.
remote: Counting objects: 100% (128/128), done.
remote: Compressing objects: 100% (75/75), done.
remote: Total 84821 (delta 75), reused 93 (delta 53), pack-reused 84693
Receiving objects: 100% (84821/84821), 132.86 MiB | 39.68 MiB/s, done.
Resolving deltas: 100% (60440/60440), done.
~/Downloads cd pdf.js
~/Downloads/pdf.js (master| ✔) npm i --force
npm WARN using --force Recommended protections disabled.
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: eslint-plugin-mozilla@2.12.4
npm WARN Found: eslint-plugin-html@7.1.0
npm WARN node_modules/eslint-plugin-html
npm WARN   dev eslint-plugin-html@"^7.1.0" from the root project
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer eslint-plugin-html@"^6.0.0" from eslint-plugin-mozilla@2.12.4
npm WARN node_modules/eslint-plugin-mozilla
npm WARN   dev eslint-plugin-mozilla@"^2.12.4" from the root project
npm WARN
npm WARN Conflicting peer dependency: eslint-plugin-html@6.2.0
npm WARN node_modules/eslint-plugin-html
npm WARN   peer eslint-plugin-html@"^6.0.0" from eslint-plugin-mozilla@2.12.4
npm WARN   node_modules/eslint-plugin-mozilla
npm WARN     dev eslint-plugin-mozilla@"^2.12.4" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: eslint-plugin-mozilla@2.12.4
npm WARN Found: eslint-plugin-prettier@4.2.1
npm WARN node_modules/eslint-plugin-prettier
npm WARN   dev eslint-plugin-prettier@"^4.2.1" from the root project
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer eslint-plugin-prettier@"^3.0.0" from eslint-plugin-mozilla@2.12.4
npm WARN node_modules/eslint-plugin-mozilla
npm WARN   dev eslint-plugin-mozilla@"^2.12.4" from the root project
npm WARN
npm WARN Conflicting peer dependency: eslint-plugin-prettier@3.4.1
npm WARN node_modules/eslint-plugin-prettier
npm WARN   peer eslint-plugin-prettier@"^3.0.0" from eslint-plugin-mozilla@2.12.4
npm WARN   node_modules/eslint-plugin-mozilla
npm WARN     dev eslint-plugin-mozilla@"^2.12.4" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: eslint-plugin-mozilla@2.12.4
npm WARN Found: prettier@2.7.1
npm WARN node_modules/prettier
npm WARN   dev prettier@"^2.7.1" from the root project
npm WARN   2 more (eslint-plugin-prettier, stylelint-prettier)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer prettier@"^1.19.1" from eslint-plugin-mozilla@2.12.4
npm WARN node_modules/eslint-plugin-mozilla
npm WARN   dev eslint-plugin-mozilla@"^2.12.4" from the root project
npm WARN
npm WARN Conflicting peer dependency: prettier@1.19.1
npm WARN node_modules/prettier
npm WARN   peer prettier@"^1.19.1" from eslint-plugin-mozilla@2.12.4
npm WARN   node_modules/eslint-plugin-mozilla
npm WARN     dev eslint-plugin-mozilla@"^2.12.4" from the root project
npm WARN deprecated fsevents@1.2.9: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated mkdirp@0.5.4: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated mkdirp@0.3.5: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated highlight.js@9.18.5: Support has ended for 9.x series. Upgrade to @latest
npm WARN deprecated core-js@2.6.9: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.

> postinstall
> cross-env PUPPETEER_PRODUCT=firefox node node_modules/puppeteer/install.js

Requesting latest Firefox Nightly version from https://product-details.mozilla.org/1.0/firefox_versions.json
Downloading Firefox Nightly r106.0a1 - 139.2 Mb [===================] 100% 0.0s
Firefox Nightly (106.0a1) downloaded to /Users/srmagura/Downloads/pdf.js/node_modules/puppeteer/.local-firefox/mac-106.0a1

added 1150 packages, and audited 1653 packages in 36s

118 packages are looking for funding
  run `npm fund` for details

26 vulnerabilities (10 moderate, 13 high, 3 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
~/Downloads/pdf.js (master|✚1) gulp dist-install
[11:27:17] Using gulpfile ~/Downloads/pdf.js/gulpfile.js
[11:27:17] Starting 'dist-install'...
[11:27:17] Starting 'dist-pre'...
[11:27:17] Starting 'generic'...
[11:27:17] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:17] Finished 'createBuildNumber' after 24 ms
[11:27:17] Starting 'locale'...

### Building localization files
[11:27:17] Finished 'locale' after 66 ms
[11:27:17] Starting 'scriptingGeneric'...

### Building default preferences
[11:27:18] asset pdf.scripting.js 105 KiB [emitted] (name: main)

webpack 5.74.0 compiled successfully
[11:27:18] Finished 'scriptingGeneric' after 691 ms
[11:27:18] Starting 'createGeneric'...

### Creating generic viewer
[BABEL] Note: The code generator has deoptimised the styling of /Users/srmagura/Downloads/pdf.js/external/quickjs/quickjs-eval.js as it exceeds the max of 500KB.
[11:27:20] asset pdf.sandbox.js 616 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:22] asset viewer.js 416 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:24] asset pdf.js 490 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:24] asset pdf.worker.js 1.88 MiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:24] Finished 'createGeneric' after 5.65 s
[11:27:24] Finished 'generic' after 6.44 s
[11:27:24] Starting 'generic-legacy'...
[11:27:24] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:24] Finished 'createBuildNumber' after 27 ms
[11:27:24] Starting 'locale'...

### Building localization files
[11:27:24] Finished 'locale' after 56 ms
[11:27:24] Starting 'scriptingGenericLegacy'...

### Building default preferences
[11:27:24] asset pdf.scripting.js 110 KiB [emitted] (name: main)

webpack 5.74.0 compiled successfully
[11:27:24] Finished 'scriptingGenericLegacy' after 232 ms
[11:27:24] Starting 'createGenericLegacy'...

### Creating generic (legacy) viewer
[BABEL] Note: The code generator has deoptimised the styling of /Users/srmagura/Downloads/pdf.js/external/quickjs/quickjs-eval.js as it exceeds the max of 500KB.
[11:27:26] asset pdf.sandbox.js 622 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:28] asset viewer.js 510 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:30] asset pdf.js 658 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:30] asset pdf.worker.js 1.98 MiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:30] Finished 'createGenericLegacy' after 6.02 s
[11:27:30] Finished 'generic-legacy' after 6.34 s
[11:27:30] Starting 'components'...
[11:27:30] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:30] Finished 'createBuildNumber' after 30 ms
[11:27:30] Starting 'createComponents'...

### Creating generic components
[11:27:30] asset pdf_viewer.js 224 KiB [emitted] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:31] Finished 'createComponents' after 455 ms
[11:27:31] Finished 'components' after 485 ms
[11:27:31] Starting 'components-legacy'...
[11:27:31] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:31] Finished 'createBuildNumber' after 32 ms
[11:27:31] Starting 'createComponentsLegacy'...

### Creating generic (legacy) components
[11:27:31] asset pdf_viewer.js 261 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:31] Finished 'createComponentsLegacy' after 493 ms
[11:27:31] Finished 'components-legacy' after 525 ms
[11:27:31] Starting 'image_decoders'...
[11:27:31] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:31] Finished 'createBuildNumber' after 25 ms
[11:27:31] Starting 'createImageDecoders'...

### Creating image decoders
[11:27:32] asset pdf.image_decoders.js 242 KiB [emitted] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:32] Finished 'createImageDecoders' after 472 ms
[11:27:32] Finished 'image_decoders' after 497 ms
[11:27:32] Starting 'image_decoders-legacy'...
[11:27:32] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:32] Finished 'createBuildNumber' after 26 ms
[11:27:32] Starting 'createImageDecodersLegacy'...

### Creating (legacy) image decoders
[11:27:32] asset pdf.image_decoders.js 291 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:32] Finished 'createImageDecodersLegacy' after 569 ms
[11:27:32] Finished 'image_decoders-legacy' after 596 ms
[11:27:32] Starting 'lib'...
[11:27:32] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:32] Finished 'createBuildNumber' after 26 ms
[11:27:32] Starting 'scriptingLib'...

### Building default preferences
[11:27:32] asset pdf.scripting.js 105 KiB [emitted] (name: main)

webpack 5.74.0 compiled successfully
[11:27:32] Finished 'scriptingLib' after 198 ms
[11:27:32] Starting 'createLib'...
[BABEL] Note: The code generator has deoptimised the styling of /Users/srmagura/Downloads/pdf.js/external/quickjs/quickjs-eval.js as it exceeds the max of 500KB.
[11:27:33] asset pdf.sandbox.js 616 KiB [emitted] [big] (name: main)

webpack 5.74.0 compiled successfully
[11:27:36] Finished 'createLib' after 3.5 s
[11:27:36] Finished 'lib' after 3.73 s
[11:27:36] Starting 'minified'...
[11:27:36] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:36] Finished 'createBuildNumber' after 43 ms
[11:27:36] Starting 'locale'...

### Building localization files
[11:27:36] Finished 'locale' after 57 ms
[11:27:36] Starting 'scriptingMinified'...

### Building default preferences
[11:27:36] asset pdf.scripting.js 105 KiB [emitted] (name: main)

webpack 5.74.0 compiled successfully
[11:27:36] Finished 'scriptingMinified' after 185 ms
[11:27:36] Starting 'createMinified'...

### Creating minified viewer
[BABEL] Note: The code generator has deoptimised the styling of /Users/srmagura/Downloads/pdf.js/external/quickjs/quickjs-eval.js as it exceeds the max of 500KB.
[11:27:38] asset pdf.sandbox.js 616 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:39] asset pdf.image_decoders.js 242 KiB [emitted] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:39] asset viewer.js 416 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:41] asset pdf.js 490 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:42] asset pdf.worker.js 1.88 MiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:42] Finished 'createMinified' after 5.66 s
[11:27:42] Starting 'compressMinified'...

### Minifying js files

### Cleaning js files
[11:27:47] Finished 'compressMinified' after 5.26 s
[11:27:47] Finished 'minified' after 11 s
[11:27:47] Starting 'minified-legacy'...
[11:27:47] Starting 'createBuildNumber'...

### Getting extension build number
Extension build number: 64
[11:27:47] Finished 'createBuildNumber' after 37 ms
[11:27:47] Starting 'locale'...

### Building localization files
[11:27:47] Finished 'locale' after 55 ms
[11:27:47] Starting 'scriptingMinifiedLegacy'...

### Building default preferences
[11:27:47] asset pdf.scripting.js 110 KiB [emitted] (name: main)

webpack 5.74.0 compiled successfully
[11:27:47] Finished 'scriptingMinifiedLegacy' after 220 ms
[11:27:47] Starting 'createMinifiedLegacy'...

### Creating minified (legacy) viewer
[BABEL] Note: The code generator has deoptimised the styling of /Users/srmagura/Downloads/pdf.js/external/quickjs/quickjs-eval.js as it exceeds the max of 500KB.
[11:27:50] asset pdf.sandbox.js 622 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:52] asset viewer.js 510 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:53] asset pdf.js 658 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:54] asset pdf.worker.js 1.98 MiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:54] asset pdf.image_decoders.js 291 KiB [emitted] [big] (name: main) 1 related asset

webpack 5.74.0 compiled successfully
[11:27:54] Finished 'createMinifiedLegacy' after 6.45 s
[11:27:54] Starting 'compressMinifiedLegacy'...

### Minifying js files

### Cleaning js files
[11:27:59] Finished 'compressMinifiedLegacy' after 5.3 s
[11:27:59] Finished 'minified-legacy' after 12 s
[11:27:59] Starting 'types'...
### Generating TypeScript definitions using `tsc`
[11:28:02] Finished 'types' after 2.57 s
[11:28:02] Starting 'createDist'...

### Cloning baseline distribution

### Overwriting all files
[11:28:04] Finished 'createDist' after 2.53 s
[11:28:04] Finished 'dist-pre' after 47 s
[11:28:04] Starting 'createDistInstall'...
Error: command "npm" with parameters "install,build/dist/" exited with code 1
[11:28:05] The following tasks did not complete: dist-install, createDistInstall
[11:28:05] Did you forget to signal async completion?
```
</details>

The cause is that `npm install build/dist/` fails with the same reason with #15429.

```
$  npm install build/dist/
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: eslint-plugin-mozilla@3.0.0
npm ERR! Found: eslint-plugin-prettier@4.2.1
npm ERR! node_modules/eslint-plugin-prettier
npm ERR!   dev eslint-plugin-prettier@"^4.2.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint-plugin-prettier@"^3.0.0" from eslint-plugin-mozilla@3.0.0
npm ERR! node_modules/eslint-plugin-mozilla
npm ERR!   dev eslint-plugin-mozilla@"^3.0.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint-plugin-prettier@3.4.1
npm ERR! node_modules/eslint-plugin-prettier
npm ERR!   peer eslint-plugin-prettier@"^3.0.0" from eslint-plugin-mozilla@3.0.0
npm ERR!   node_modules/eslint-plugin-mozilla
npm ERR!     dev eslint-plugin-mozilla@"^3.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/tamura/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tamura/.npm/_logs/2022-11-20T02_03_39_718Z-debug-0.log
```

One of  options is adding `--force` to the line:

https://github.com/mozilla/pdf.js/blob/ae7c97aef81aa2934fbbbdc4a5511e0b274e086a/gulpfile.js#L2318

Another option is adding `overrides` to `package.json`:

```
  "overrides": {
    "eslint-plugin-prettier": "^4.2.1",
    "prettier": "^2.7.1"
  },
```

See

> If you need to make specific changes to dependencies of your dependencies, for example replacing the version of a dependency with a known security issue, replacing an existing dependency with a fork, or making sure that the same version of a package is used everywhere, then you may add an override.

- https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

A bonus is that, with the new property, `npm install` does not fail either.

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -2345,7 +2345,7 @@ gulp.task(
       opts.cwd = installPath;
       distPath = path.relative(installPath, distPath);
     }
-    safeSpawnSync("npm", ["install", distPath], opts);
+    safeSpawnSync("npm", ["install", "--force", distPath], opts);
     done();
   })
 );

